### DISCUSSION TITLE
Hide HTML/XML tags in preview

### DISCUSSION THREAD

**Discussion by MaxGyver83** [spacecowboy/Feeder#248] [Ideas]
In one of my feeds, I often see raw HTML/XML in the preview. Example:

![Feeder_screenshot_cropped](https://github.com/spacecowboy/Feeder/assets/13970628/8394ddba-521e-4f9c-895f-2a980130c7de)

It would be nice to skip this part (`<!CDATA[<p> <a href="https://www.tages...>`) and show the first readable words of this news instead. What do you think?

This is the feed in case you want to try yourself:

<https://www.tagesschau.de/xml/atom/>

**Comment by spacecowboy** [comment]
It's just a bug.

It can be tricky to decode all feeds correctly

**Reply by spacecowboy** [reply]
In this case, the feed is actually doing something very wrong by first wrapping their HTML in CDATA tags and then escaping that - effectively escaping it twice.

**Reply by spacecowboy** [reply]
but it shows correctly in Reader?

**Reply by MaxGyver83** [reply]
> but it shows correctly in Reader?

Yes. This is the beginning of the first news in the other screenshot:

![Screenshot_2024-03-24-14-52-27-92_56dde218d5b132d57db5f243d6cd7745~(1).png](https://github.com/spacecowboy/Feeder/assets/13970628/daa94345-db1e-4f3e-8a86-5a7592a8ff52)

**Reply by MaxGyver83** [reply]
> In this case, the feed is actually doing something very wrong by first wrapping their HTML in CDATA tags and then escaping that - effectively escaping it twice.

Maybe I can tell them. But probably they do it for a reason. Maybe a workaround for a different issue!?

**Reply by spacecowboy** [reply]
If it shows correctly in the reader, then the app is doing something different for the preview

**Reply by spacecowboy** [reply]
Note to self: Use JSOUP for plaintext conversion as well

**Comment by yoshimo** [comment]
I was about to File a Bug Report for this Problem, whatsbthe current state ? Did you get  a reply from the Website Team?

**Reply by MaxGyver83** [reply]
@spacecowboy wrote that the fix is to use JSOUP for plaintext conversion as well. (See the last comment before yours.)

**Reply by yoshimo** [reply]
The ARD could change their Feed too, was wondering If you asked for that.

**Reply by spacecowboy** [reply]
in this case their feed is perfectly valid. there is nothing for them to change

**Comment by spacecowboy** [comment]
Someone should make a proper bug report for this. Having it in conversations means it will be forgotten

**Reply by yoshimo** [reply]
#819
